lib/DeveloperStatsCommander.js

```diff
@@ -0,0 +1,234 @@
+
+        inquirer.prompt([
+            {
+                type: 'input',
+                name: 'org',
+                message: 'Name of your organization on github.com',
+                default: process.env.DEFAULT_ORGANIZATION
+            }
+        ], function(answers) {
+            deferred.resolve(answers.org);
+        });
+
+        return deferred.promise;
+    };
+
+    DeveloperStatsCommander.prototype.digest = function (resolves) {
+
+        var values = this.extractAndEnsureValues (resolves);
+
+        var digestedContributorStats = {}; 
+        var digestedRepoStats = {}; 
+
+        for (var i = 0 ; i < values.length; i++) {
+
+            var originalStatsPayload = values [i];
+            var repo = originalStatsPayload.repo;
+
+            for (var p in originalStatsPayload) {
+
+                if (isNaN (p)) {
+                    continue;
+                }
+
+                var weeklyStats = originalStatsPayload [p];
+
+                for (var w = 0; w < weeklyStats.weeks.length; w++) {
+
+                    var statsTimestamp = weeklyStats.weeks[w].w;
+                    var thisWeekTimestamp = moment().startOf('week').add(1, 'hour').format('X');
+
+                    if (statsTimestamp != thisWeekTimestamp) {
+                        continue;
+                    }
+
+                    if (typeof digestedContributorStats [weeklyStats.author.login] == 'undefined') {
+                        digestedContributorStats [weeklyStats.author.login] = {};
+                    }
+
+                    if (typeof digestedRepoStats [repo] == 'undefined') {
+                        digestedRepoStats [repo] = {};
+                    }
+
+                    var entry = {
+                        commits: weeklyStats.weeks[w].c,
+                        added : weeklyStats.weeks[w].a,
+                        deleted : weeklyStats.weeks[w].d,
+                        changed : weeklyStats.weeks[w].a + weeklyStats.weeks[w].d
+                    };
+
+                    if (entry.commits == 0) {
+                        continue;
+                    }
+
+                    digestedContributorStats [weeklyStats.author.login][repo] = entry;
+                    digestedRepoStats [repo][weeklyStats.author.login] = entry;
+                }
+            }
+        }
+
+        var repoStats = this.snitch (digestedRepoStats);
+        var contributorStats = this.snitch (digestedContributorStats);
+
+        console.log (repoStats);
+        console.log (contributorStats);
+
+        if (process.env.SLACK_ENABLE == 'true') {
+            this.slack (repoStats);
+            this.slack (contributorStats);
+        }
+
+        return;
+    }
+
+    DeveloperStatsCommander.prototype.slack = function (message) {
+
+        var sh = 'curl -X POST --data-urlencode \'payload={"username": "' + process.env.SLACK_USERNAME + '", "text": "'+ message + '", "icon_emoji": "' + process.env.SLACK_EMOJI + '"}\' ' + process.env.SLACK_URL;
```

Comment:
You can use http://sugarjs.com/api/String/assign, to make this a bit more readable.